api/mappers/dynamodb.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, Dict, List, TypeAlias, TypeVar, Union
+
+from pydantic import BaseModel
+
+from mappers.engine import (
+    map_environment_api_key_to_engine,
+    map_environment_to_engine,
+    map_identity_to_engine,
+)
+
+if TYPE_CHECKING:  # pragma: no cover
+    from flag_engine.identities.models import IdentityModel
+
+    from environments.identities.models import Identity
+    from environments.models import Environment, EnvironmentAPIKey
+
+
+__all__ = (
+    "map_engine_identity_to_identity_document",
+    "map_environment_api_key_to_environment_api_key_document",
+    "map_environment_to_environment_document",
+    "map_identity_to_identity_document",
+)
+
+DocumentValue: TypeAlias = Union[Dict[str, "DocumentValue"], str, bool, None, Decimal]
+Document: TypeAlias = Dict[str, DocumentValue]
+
+
+def map_environment_to_environment_document(
+    environment: "Environment",
+) -> Document:
+    return {
+        field_name: _map_value_to_document_value(value)
+        for field_name, value in map_environment_to_engine(
+            environment,
+        )
+    }
+
+
+def map_environment_api_key_to_environment_api_key_document(
+    environment_api_key: "EnvironmentAPIKey",
+) -> Document:
+    return {
+        field_name: _map_value_to_document_value(value)
+        for field_name, value in map_environment_api_key_to_engine(environment_api_key)
+    }
+
+
+def map_engine_identity_to_identity_document(
+    engine_identity: "IdentityModel",
+) -> Document:
+    return {
+        field_name: _map_value_to_document_value(value)
+        for field_name, value in engine_identity
+    }
+
+
+def map_identity_to_identity_document(
+    identity: "Identity",
+) -> Document:
+    return map_engine_identity_to_identity_document(map_identity_to_engine(identity))
+
+
+T = TypeVar("T")
```

Comment:
Is there a reason to define this here vs with the other type definitions at the top of the module?

Reply:
None except to emphasize that it's used exclusively by `_noop_encoder`.